File: android/versioned-abis/expoview-abi42_0_0/src/main/java/abi42_0_0/expo/modules/sensors/services/AccelerometerService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package abi42_0_0.expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import abi42_0_0.org.unimodules.core.interfaces.InternalModule;

import abi42_0_0.expo.modules.interfaces.sensors.services.AccelerometerServiceInterface;

public class AccelerometerService extends SubscribableSensorService implements InternalModule, AccelerometerServiceInterface {
  public AccelerometerService(Context reactContext) {
    super(reactContext);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_ACCELEROMETER;
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(AccelerometerServiceInterface.class);
  }
}

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/AccelerometerService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import expo.modules.interfaces.sensors.services.AccelerometerServiceInterface
import expo.modules.core.interfaces.InternalModule

class AccelerometerService(reactContext: Context?) : SubscribableSensorService(reactContext), InternalModule, AccelerometerServiceInterface {
  override val sensorType: Int = Sensor.TYPE_ACCELEROMETER

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(AccelerometerServiceInterface::class.java)
  }
}
